src/Util/FilePath.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Util;
+
+use Rector\Configuration\Option;
+use Rector\Configuration\Parameter\SimpleParameterProvider;
+
+/**
+ * @see \Rector\Tests\Util\FilePathTest
+ */
+final class FilePath
+{
+    public static function fileIsInRectorPathOrSource(string $file): bool
```

Comment:
The path logic may need to similar with 

https://github.com/rectorphp/rector-src/blob/ce8575f35a2ea096e91223347074542c1e603742/src/Configuration/ConfigurationFactory.php#L140

I will try if I can create alternative PR with `DynamicSourceLocatorProvider` per 

https://github.com/rectorphp/rector-src/pull/6890#pullrequestreview-2834334643

**when I have a chance :)**

This case is interesting of how locate identifier can works in the future imo :)

Reply:
see alternative PR:

- https://github.com/rectorphp/rector-src/pull/6899 

:)